### PR TITLE
systemd: 249.5 -> 249.7 & various fixes

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
This is a rebase version of #121371 + the bump to 249.7

cc @arianvp @flokli 

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/systemd/default.nix [hunk 1]
```diff
@@ -166,14 +166,6 @@ stdenv.mkDerivation {
     # all our root unit dirs if they are symlinks. This does exactly what we
     # need (AFAICT).
     ./0019-core-handle-lookup-paths-being-symlinks.patch
```
Comment (flokli) [inline pkgs/os-specific/linux/systemd/default.nix:168]: Minor nit: We should add a link to https://github.com/systemd/systemd/pull/20479 here, so the PR sent upstream can be found easier.

**Review by flokli — COMMENTED** [review]
Build queued ;-)

**Comment by flokli** [comment]
Let's cook this in staging a bit :-)

**Comment by yu-re-ka** [comment]
Should this upgrade the running init process when running rebuild switch?

```
+ ssh -p62954 yaya.yuka.dev 'sudo /nix/var/nix/profiles/system/bin/switch-to-configuration switch'
updating GRUB 2 menu...
removing obsolete file /boot/kernels/nranbvnqg5ijmjrnvj6g5fg2w90ia3ha-initrd-linux-5.15.4-initrd
removing obsolete file /boot/kernels/shl48yrwmdxnndrmim11ylbvpsfzjjc3-linux-5.15.4-bzImage
stopping the following units: acme-fixperms.service, audit.service, bind.service, coturn.service, deluged.service, dhparams-init.service, docker.service, docker.socket, dovecot2.service, gitea-port-forward.service, gitea.service, gotify-server.service, hedgedoc.service, heisenbridge.service, iperf3.service, jellyfin.service, jicofo.service, jitsi-videobridge2.service, kmod-static-nodes.service, matrix-synapse.service, mautrix-telegram.service, mount-pstore.service, network-local-commands.service, nix-daemon.service, nix-daemon.socket, nscd.service, opendkim.service, pleroma.service, postfix.service, postgresql.service, proc-sys-fs-binfmt_misc.automount, prometheus-jitsi-exporter.service, prometheus-nginx-exporter.service, prometheus-node-exporter.service, prosody.service, radicale.service, redis.service, restic-rest-server.service, rspamd.service, systemd-binfmt.service, systemd-modules-load.service, systemd-networkd-wait-online.service, systemd-networkd.service, systemd-networkd.socket, systemd-resolved.service, systemd-sysctl.service, systemd-timesyncd.service, systemd-tmpfiles-clean.timer, systemd-tmpfiles-setup-dev.service, systemd-udev-trigger.service, systemd-udevd-control.socket, systemd-udevd-kernel.socket, systemd-udevd.service, trainsearch.service, up-rewrite-proxy.service, vnstat.service, zram-reloader.service
NOT restarting the following changed units: getty@tty1.service, systemd-fsck@dev-disk-by\x2duuid-95271c9f\x2dcd5a\x2d4c29\x2da2c8\x2da7b22ac95184.service, systemd-fsck@dev-disk-by\x2duuid-9a4eac20\x2d0c82\x2d4793\x2d9537\x2d4970273d7204.service, systemd-journal-flush.service, systemd-logind.service, systemd-random-seed.service, systemd-remount-fs.service, systemd-tmpfiles-setup.service, systemd-update-utmp.service, systemd-user-sessions.service, user-runtime-dir@1000.service, user@1000.service, zram-init-zram0.service
activating the configuration...
setting up secrets...
setting up /etc...
setting up secrets...
restarting systemd...
reloading user units for yuka...
setting up tmpfiles
reloading the following units: dbus.service, dev-hugepages.mount, dev-mqueue.mount, gitlab-runner.service, nftables.service, proc-sys-fs-binfmt_misc.mount, reload-systemd-vconsole-setup.service, sys-fs-fuse-connections.mount, sys-kernel-config.mount, sys-kernel-debug.mount
gitlab-runner.service is not active, cannot reload.
proc-sys-fs-binfmt_misc.mount is not active, cannot reload.
restarting the following units: home-manager-yuka.service, nginx.service, sshd.service, systemd-journald.service
starting the following units: acme-fixperms.service, audit.service, bind.service, coturn.service, deluged.service, dhparams-init.service, docker.socket, dovecot2.service, gitea-port-forward.service, gitea.service, gotify-server.service, hedgedoc.service, heisenbridge.service, iperf3.service, jellyfin.service, jicofo.service, jitsi-videobridge2.service, kmod-static-nodes.service, matrix-synapse.service, mautrix-telegram.service, mount-pstore.service, network-local-commands.service, nix-daemon.socket, nscd.service, opendkim.service, pleroma.service, postfix.service, postgresql.service, proc-sys-fs-binfmt_misc.automount, prometheus-jitsi-exporter.service, prometheus-nginx-exporter.service, prometheus-node-exporter.service, prosody.service, radicale.service, redis.service, restic-rest-server.service, rspamd.service, systemd-binfmt.service, systemd-modules-load.service, systemd-networkd-wait-online.service, systemd-networkd.socket, systemd-resolved.service, systemd-sysctl.service, systemd-timesyncd.service, systemd-tmpfiles-clean.timer, systemd-tmpfiles-setup-dev.service, systemd-udev-trigger.service, systemd-udevd-control.socket, systemd-udevd-kernel.socket, trainsearch.service, up-rewrite-proxy.service, vnstat.service, zram-reloader.service
the following new units were started: systemd-vconsole-setup.service
```

**Comment by dasJ** [comment]
@yu-re-ka It does: `restarting systemd...`

**Comment by yu-re-ka** [comment]
Somehow didn't notice that line, everything is good then :)

**Pull request merged**